runtime/src/message_processor.rs

```diff
@@ -832,7 +843,13 @@ impl MessageProcessor {
                 Self::create_keyed_accounts(message, instruction, executable_accounts, accounts);
```

Comment:
This mess right here will be reimplemented differently in a [subsequent PR](https://github.com/solana-labs/solana/pull/15720). So polishing it would be somewhat pointless.

Reply:
Can you give more details on what you think the mess so it can be addressed in future prs?

Reply:
There is some `unsafe` code to erase life times as we can not tell rust that the `keyed_accounts` created here will exist only in this VM stack frame. But if we only recycle existing `keyed_accounts` this could be avoided, I think.